addons/sale/static/src/scss/sale_portal.scss

```diff
@@ -8,3 +8,8 @@
 .orders_label_text_align {
     vertical-align: 15%;
 }
+
+.oe_sale_order_tax_table {
+    margin-bottom: 0;
+    margin-top: -1rem;
```

Comment:
Why is it needed?

Reply:
margin-top -1rem is needed because default Bootstrap table or table-sm classes have a margin-bottom of 1rem. This is just for compensating (not really elegant though)